src/backend/admin_ops/users.py

```diff
@@ -0,0 +1,240 @@
+# Copyright 2023 SUSE LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+from typing import Any, Dict, List, cast
+
+from pydantic import parse_obj_as
+
+from backend.admin_ops import do_request
+from backend.admin_ops.errors import MissingParameterError
+from backend.admin_ops.types import (
+    AuthUser,
+    UserInfo,
+    UserKeyOpParams,
+    UserKeys,
+    UserOpParams,
+    UserQuotaOpParams,
+    model_to_params,
+)
+
+
+async def get_user_info(
+    url: str,
+    access_key: str,
+    secret_key: str,
+    uid: str | None = None,
+    with_statistics: bool = False,
+) -> UserInfo:
+    """
+    Obtain informations about the user to whom the `access_key:secret_key` pair
+    belongs to.
+    """
+    params = {"access-key": access_key, "stats": with_statistics}
+    if uid is not None and len(uid) > 0:
+        params["uid"] = uid
+
+    res = await do_request(
+        url=url,
+        access_key=access_key,
+        secret_key=secret_key,
+        endpoint="/admin/user",
+        method="GET",
+        params=params,
+    )
+    return UserInfo.parse_obj(res.json())
+
+
+async def authenticate(url: str, access_key: str, secret_key: str) -> AuthUser:
```

Comment:
I don't have a clear idea why we need this function.
S3 calls are RESTful apis. 
If we are assuming that a certain user after a successful `authenticate` function will be considered authenticated forever, this is wrong.
If this is needed just to check that the user can access the admin ops set, that is fair.
In the latter case (that is suspect is the case), I would change the function with a name such as: "is_admin_user".
(We are not actually authenticating the user, we are just checking if he can perform admin ops).

Reply:
this is meant to be used to ensure the user can authenticate with the admin ops API. The function can definitely be renamed to reflect a bit better its intended purpose.